homeassistant/components/enigma2/media_player.py

```diff
@@ -0,0 +1,251 @@
+
+    @property
+    def media_content_type(self):
+        """Type of video currently playing."""
+        return MEDIA_TYPE_TVSHOW
+
+    @property
+    def is_volume_muted(self):
+        """Boolean if volume is currently muted."""
+        return self.muted
+
+    @property
+    def media_image_url(self):
+        """Picon url for the channel."""
+        return self.picon_url
+
+    def set_volume_level(self, volume):
+        """Set volume level, range 0..1."""
+        self.e2_box.set_volume(int(volume * 100))
+
+    def volume_up(self):
+        """Volume up the media player."""
+        self.e2_box.set_volume(self.volume + 5)
+
+    def volume_down(self):
+        """Volume down media player."""
+        self.e2_box.set_volume(self.volume - 5)
+
+    @property
+    def volume_level(self):
+        """Volume level of the media player (0..1)."""
+        return self.volume / 100
+
+    def media_stop(self):
+        """Send stop command."""
+        self.e2_box.set_stop()
+
+    def media_play(self):
+        """Play media."""
+        self.e2_box.toggle_play_pause()
+
+    def media_pause(self):
+        """Pause the media player."""
+        self.e2_box.toggle_play_pause()
+
+    def media_next_track(self):
+        """Send next track command."""
+        self.e2_box.set_channel_up()
+
+    def media_previous_track(self):
+        """Send next track command."""
+        self.e2_box.set_channel_down()
+
+    def mute_volume(self, mute):
+        """Mute or unmute."""
+        self.e2_box.mute_volume()
+
+    @property
+    def source(self):
+        """Return the current input source."""
+        return self.current_service_channel_name
+
+    @property
+    def source_list(self):
+        """List of available input sources."""
+        return list(self.sources.keys())
+
+    @asyncio.coroutine
+    def async_select_source(self, source):
+        """Select input source."""
+        self.e2_box.select_source(self.sources[source])
+
+    def update(self):
+        """Update state of the media_player."""
+        status_info = self.e2_box.refresh_status_info()
+        if self.e2_box.is_box_in_standby():
+            self._state = STATE_OFF
+            return
+        self.current_service_channel_name = \
+            status_info['currservice_station']
+        pname = status_info['currservice_name']
+        self.current_programme_name = pname if pname != "N/A" else ""
+        self.current_service_ref = status_info['currservice_serviceref']
+        self.status_info = status_info
+        if self.current_service_ref.startswith('1:0:0'):
+            self._state = STATE_PLAYING
+        else:
+            self._state = STATE_ON
+        self.muted = status_info['muted']
+        self.volume = status_info['volume']
+        self.picon_url = \
+            self.e2_box.get_current_playing_picon_url(
+                channel_name=self.current_service_channel_name,
+                currservice_serviceref=self.current_service_ref)
```

Comment:
Having no idea about how this is handled by the backend, it would be nicer API-wise to avoid passing those parameters if the function has access to that information already.

Reply:
Done now in the pypi module.